app/models/manageiq/providers/redhat/infra_manager/vm/operations/snapshot.rb

```diff
@@ -1,4 +1,18 @@
 module ManageIQ::Providers::Redhat::InfraManager::Vm::Operations::Snapshot
+  extend ActiveSupport::Concern
+
+  included do
+    supports :snapshots do
+      unless ext_management_system&.supports_snapshots?
+        if ext_management_system
```

Comment:
>  I would like to make a PR that will allow one to write something like supports :snapshots, through: ext_management_system kinda like cancan syntax.

@borod108 we were thinking about something like this too, but I think we should hold off until we see a clearer picture of how this is employed. I.e. by adding more flexibility, like delegation, to the DSL we might in fact solving only a symptom and not the root cause (operations API).

But huge thanks for pointing this out and I'll try to keep in the loop of how that goes along

cc @blomquisg @Ladas